x-pack/plugins/security/public/nav_control/nav_control_service.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Subscription } from 'rxjs';
+import { CoreStart } from 'src/core/public';
+import ReactDOM from 'react-dom';
+import React from 'react';
+import { SecurityLicense } from '../../common/licensing';
+import { AuthenticatedUser } from '../../common/model';
+import { SecurityNavControl } from './nav_control_component';
+
+interface SetupDeps {
+  securityLicense: SecurityLicense;
+}
+
+interface StartDeps {
+  core: CoreStart;
+}
+
+export class SecurityNavControlService {
+  private securityLicense!: SecurityLicense;
+
+  private navControlRegistered!: boolean;
+
+  private securityFeaturesSubscription?: Subscription;
+
+  public setup({ securityLicense }: SetupDeps) {
+    this.securityLicense = securityLicense;
+  }
+
+  public start({ core }: StartDeps) {
+    this.securityFeaturesSubscription = this.securityLicense.features$.subscribe(
+      ({ showLinks }) => {
+        const isAnonymousPath = core.http.anonymousPaths.isAnonymous(window.location.pathname);
+
+        const shouldRegisterNavControl =
+          !isAnonymousPath && showLinks && !this.navControlRegistered;
+
+        if (shouldRegisterNavControl) {
+          const user = core.http.get('/api/security/v1/me', {
+            headers: {
+              'kbn-system-api': true,
+            },
+          }) as Promise<AuthenticatedUser>;
+          this.registerSecurityNavControl(core, user);
+        }
+      }
+    );
+  }
+
+  public stop() {
+    if (this.securityFeaturesSubscription) {
+      this.securityFeaturesSubscription.unsubscribe();
+      this.securityFeaturesSubscription = undefined;
+    }
+    this.navControlRegistered = false;
+  }
+
+  private registerSecurityNavControl(
+    core: Pick<CoreStart, 'chrome' | 'http' | 'i18n' | 'application'>,
+    user: Promise<AuthenticatedUser>
+  ) {
+    core.chrome.navControls.registerRight({
+      order: 2000,
+      mount: (el: HTMLElement) => {
```

Comment:
question: i'm not seeing an easy way to test the `mount` method, are you aware of one?

Reply:
I'm not aware of one, but the functional test suites cover this functionality already by virtue of testing the popover and links inside (account management and logout)

Reply:
Could mock `core.chrome.navControls.registerRight` to get the calling arguments, and use enzime/jsdom to test the `MountPoint`

Reply:
Something like

https://github.com/elastic/kibana/blob/01bbbf4b0391b1ded6d9e1e5dae1ee8f9e0e06f1/src/core/public/overlays/modal/modal_service.test.tsx#L47-L58